openedx/core/djangoapps/content/course_metadata/models.py

```diff
@@ -0,0 +1,40 @@
+from django.db import models
+
+from model_utils.models import TimeStampedModel
+from xmodule_django.models import CourseKeyField
+from api_manager.courseware_access import get_course_leaf_nodes
+
+
+class CourseAggregatedMetaData(TimeStampedModel):
+    """
+    Model for storing and caching aggregated metadata about a course.
+
+    This model contains aggregated metadata about a course such as
+    total modules, total assessments.
+    """
+    id = CourseKeyField(db_index=True, primary_key=True, max_length=255)  # pylint: disable=invalid-name
```

Comment:
I can't remember if primary_key=True implies a unique constraint -- if it does not we'll need to ensure uniqueness.